src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -1331,6 +1357,40 @@ enum UpdateModificationAction {
         MOVE, COPY
     }
 
+    private String getIds(UUID studyUuid, UUID nodeUuid, List<String> substationsIds, boolean inUpstreamBuiltParentNode, StudyConstants.EquipmentType equipmentType) {
+        switch (equipmentType) {
```

Comment:
i think this method should be in study service

Reply:
done